components/brave_news/browser/resources/shared/Context.tsx

```diff
@@ -146,6 +155,28 @@ export function BraveNewsContextProvider(props: { children: React.ReactNode }) {
     configurationCache.set({ openArticlesInNewTab: inNewTab })
   }, [])
 
+  const reportViewCount = (newViews: number) => {
+    getBraveNewsController().onNewCardsViewed(newViews)
+  }
+
+  const reportVisit = (depth: number) => {
+    getBraveNewsController().onCardVisited(depth + 1)
+  }
+
+  const reportSessionStart = () => {
+    getBraveNewsController().onInteractionSessionStarted()
+  }
+
+  const isSidebarFilterUsed = useRef<boolean>(false);
+  const reportSidebarFilterUsage = useCallback(() => {
```

Comment:
This could also probably be a module method:

```ts
let sidebarUsed = false
export const reportSidebarFilterUsage = () => {
  if (sidebarUsed) return
  sidebarUsed = true
  getBraveNewsController().onSidebarFilterUsage()
}

Reply:
related to https://github.com/brave/brave-core/pull/21639#discussion_r1480530414